src/core/serializer.rs

```diff
@@ -0,0 +1,51 @@
+//! Serialization of Rust data structures.
```

Comment:
If we provide the `Serializer` trait then we should use it in our `Serialize` trait.

Reply:
@Xavrax do we need `Serialize` trait? For example, we don't have `Deserialize` trait.

Reply:
For now it is not needed. 
Imagine the subscription. Then we have to deal with unknown type of data that user can provide. Then we will need that `deserialize` trait.

Reply:
@Xavrax so you suggest moving `Serializer` trait inside `Serialize` file?